infrastructure/modules/cluster/diagnostics.tf

```diff
@@ -0,0 +1,43 @@
+resource "azurerm_storage_account" "logs" {
+  count                    = var.log_cluster_diagnostics ? 1 : 0
+  name                     = "logs"
+  resource_group_name      = var.resource_group_name
+  location                 = var.location
+  account_kind             = "StorageV2"
+  account_tier             = "Standard"
+  account_replication_type = "RAGRS"
+  access_tier              = "Cool"
+
+  tags = {
+    environment = var.environment
+  }
+}
+
+variable "diagnostic_log_types" {
+  description = "Set of log types to create configuration for"
+  type        = list(string)
+  default = ["kube-apiserver",
+    "kube-audit",
+    "kube-controller-manager",
+    "kube-scheduler",
+  "cluster-autoscaler"]
+}
```

Comment:
Probably want to keep this variable in `variables.tf` just for consistency

Reply:
Good point! Moved.